src/csharp/NativeMethods.cs

```diff
@@ -171,6 +177,25 @@ internal class NativeLib
                                                                               int /* int32_t */ token,
                                                                               out IntPtr /* const char** */ outStr);
 
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern unsafe IntPtr /* OgaResult* */ OgaCreateTensorFromBuffer(IntPtr /* data* */ data, long* shape_dims, UIntPtr shape_dims_count, ElementType element_Type,
```

Comment:
Replace two shape arguments to long[] and pass it directly, no pinning required.

Reply:
How do I change it? C# knows how to split the long[] into the separate parts for a C API?

Reply:
Keep both arguments but replace the first one with long[]. Remove unsafe keyword.